tests/unit/utils/test_master.py

```diff
@@ -0,0 +1,106 @@
+            data = master.read_proc_file('x', {})
+            assert data is None
+
+    def test_read_proc_not_dict(self):
+        proc_data = '}{'
+        m_fopen = mock_open()
+        with patch('salt.payload.Serial.load', return_value=proc_data):
+            with patch('salt.utils.files.fopen', m_fopen):
+                data = master.read_proc_file('/tacos', {})
+                assert data is None
+
+    def test_read_proc_no_pid(self):
+        m_fopen = mock_open()
+        proc_data = {"tacos": "broseph"}
+        with patch('salt.payload.Serial.load', return_value=proc_data):
+            with patch('salt.utils.files.fopen', m_fopen):
+                data = master.read_proc_file('/tacos', {})
+                assert data is None
+
+
+@skipIf(not HAS_PSUTIL, "psutil needed to run test")
+class MasterUtilsIsPidHealthyPsUtil(TestCase):
+
+    def tests_pid_not_running(self):
+        assert master.is_pid_healthy(99999999) is False
+
+    def test_is_pid_healthy_running_salt(self):
+        with patch('psutil.Process.cmdline', return_value=['salt']):
+            assert master.is_pid_healthy(1) is True
+
+    def test_is_pid_healthy_not_running_salt(self):
+        with patch('psutil.Process.cmdline', return_value=['tacos']):
+            assert master.is_pid_healthy(1) is False
+
+    def tets_is_pid_healthy_raises(self):
+        with patch("psutil.Process", side_effect=psutil.NoSuchProcess):
+            assert master.is_pid_healthy(1) is False
+
+
+@patch("salt.utils.master.HAS_PSUTIL", False)
+class MasterUtilsIsPidHealthy(TestCase):
+
+    def test_is_pid_healthy_unsupported_platform(self):
+        with patch("salt.utils.platform.is_aix", return_value=True):
+            assert master.is_pid_healthy(11) is True
+
+        with patch("salt.utils.platform.is_windows", return_value=True):
+            assert master.is_pid_healthy(1) is True
+
+    def test_pid_not_running(self):
+        assert master.is_pid_healthy(99999999) is False
+
+    def test_is_pid_healthy_running_salt(self):
+        m_fopen = mock_open(read_data=b'salt')
+        with patch('salt.utils.process.os_is_running', return_value=True):
+            with patch('salt.utils.files.fopen', m_fopen):
+                assert master.is_pid_healthy(12345) is True
+
+    def test_is_pid_healthy_not_running_salt(self):
+        m_fopen = mock_open(read_data=b'tacos')
+        with patch('salt.utils.process.os_is_running', return_value=True):
+            with patch('salt.utils.files.fopen', m_fopen):
+                assert master.is_pid_healthy(12345) is False
+
+    def tets_is_pid_healthy_raises(self):
+        m_fopen = mock_open(side_effect=IOError)
+        with patch('salt.utils.files.fopen', m_fopen):
+            assert master.is_pid_healthy(12345) is False
+
+        m_fopen = mock_open(side_effect=OSError)
+        with patch('salt.utils.files.fopen', m_fopen):
+            assert master.is_pid_healthy(12345) is False
```

Comment:
Do these pytest assertion styles work well on failures with `pytest-salt`? It seems like they should, though I haven't tried it, and have typically just gone with the UnitTest style assertions. I'm not sure if we have a guideline/standard there.

Reply:
afaik, they work just fine. i did it this way bc its force of habit for me. haha. at least nothing blew up saying it didnt know how to run them. 🤷‍♂️